launcher/modplatform/helpers/HashUtils.cpp

```diff
@@ -35,6 +35,18 @@ Hasher::Ptr createFlameHasher(QString file_path)
     return new FlameHasher(file_path);
 }
 
+Hasher::Ptr createBlockedModHasher(QString file_path, ModPlatform::Provider provider)
+{
+    return new BlockedModHasher(file_path, provider);
+}
+
+Hasher::Ptr createBlockedModHasher(QString file_path, ModPlatform::Provider provider, QString type)
+{
+    auto hasher = new BlockedModHasher(file_path, provider);
```

Comment:
```suggestion
    auto hasher = createBlockedModHasher(file_path, provider);
```

this way we avoid having to change two places at once if we need to change something in there :)

Reply:
the `useHashType` method used to pass along the specific hash type isn't part of the `Hasher::ptr` interface so the type narrowing that results from using the builder prevents it's use.

Reply:
eh ok then, it's kinda weird that the previous class hierarchy uses subclassing for different providers, while here we use type definition to change this behavior. welp